doc/guides/debugging.md

```diff
@@ -0,0 +1,177 @@
+by default `5858`. Any debugger client which speaks this protocol can connect
+to and debug the running process.
+
+A running Node.js process can also be instructed to start listening for
+debugging messages by signaling it with `SIGUSR1` (on Linux and OS X).
+
+When started with the **--inspect** or **--inspect-brk** switches, Node.js
+listens for diagnostic commands defined by the [Chrome Debugging Protocol][]
+via websockets, by default at `ws://127.0.0.1:9229/node`. Any diagnostics
+client which speaks this protocol can connect to and debug the running process.
+
+The `--inspect` option and protocol are _experimental_ and may change.
+
+[V8 Debugging Protocol]: https://github.com/v8/v8/wiki/Debugging-Protocol
+[Chrome Debugging Protocol]: https://developer.chrome.com/devtools/docs/debugger-protocol
+
+
+## Debugging Clients
+
+Several commercial and open source tools can connect to Node's
+debugger and/or inspector. Info on these follows.
+
+* [Built-in Debugger](https://github.com/nodejs/node/blob/master/lib/_debugger.js)
+
+  * Start `node debug script_name.js` to start your script under Node's
+    builtin command-line debugger. Your script starts in another Node
+    process started with the `--debug-brk` option, and the initial Node
+    process runs the `_debugger.js` script and connects to your target.
+
+* [Chrome DevTools](https://github.com/ChromeDevTools/devtools-frontend)
+
+  * **Option 1 (Chrome 55+)**: Open `chrome://inspect` in your Chromium
+    browser. Click the Configure button and ensure your target host and port
+    are listed.
+  * **Option 2**: Paste the following URL in Chrome,
+    replacing the `ws` parameter with your hostname and port as needed:
+
+    `chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=localhost:9229/node`
+
+* [VS Code](https://github.com/microsoft/vscode)
+  * **Option 1**: In the Debug panel, click the settings button to choose
+    initial debug configurations. Select "Node.js v6.3+ (Experimental)" for
+    `--inspect` support, or "Node.js" for `--debug` support.
+  * **Option 2**: In debug configurations in `.vscode/launch.json`, set `type`
+    as `node2`.
+
+  * For more info, see <https://github.com/Microsoft/vscode-node-debug2>.
+
+* [WebStorm](https://www.jetbrains.com/webstorm/) 2016.2+ and other JetBrains
+                                                  IDEs (IJ U, PS, PC, RM, CL, AC)
+  * In the Node.js debug configuration add `--inspect` to the Node parameters
+    field for `--inspect` support. By default, the debug process will use V8
+    Debugging Protocol.
+
+* [node-inspector](https://github.com/node-inspector/node-inspector)
+
+* [chrome-remote-interface](https://github.com/cyrus-and/chrome-remote-interface)
+
+
+## Debugging command-line options
+
+* The following table clarifies the implications of various diag-related flags:
+
+<table cellpadding=0 cellspacing=0>
```

Comment:
Seeing all this HTML makes me cringe a bit. What about using [github's table markdown syntax](https://help.github.com/articles/organizing-information-with-tables/)?